webpack/set-webpack-public-path-plugin/src/SetPublicPathPlugin.ts

```diff
@@ -96,22 +97,24 @@ interface IAsset {
   source(): string;
 }
 
-interface IExtendedMainTemplate {
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare const __dummyWebpack4MainTemplate: Webpack4.compilation.MainTemplate;
+interface IWebpack4ExtendedMainTemplate extends Webpack4.compilation.MainTemplate {
   hooks: {
-    jsonpScript?: Tapable.SyncWaterfallHook<string, Webpack.compilation.Chunk, string>;
-    requireExtensions: Tapable.SyncWaterfallHook<string, Webpack.compilation.Chunk, string>;
-    startup: Tapable.SyncHook<string, Webpack.compilation.Chunk, string>;
-  };
+    jsonpScript?: Tapable.SyncWaterfallHook<string, Webpack4.compilation.Chunk, string>;
+    requireExtensions: Tapable.SyncWaterfallHook<string, Webpack4.compilation.Chunk, string>;
+    startup: Tapable.SyncHook<string, Webpack4.compilation.Chunk, string>;
+  } & typeof __dummyWebpack4MainTemplate.hooks;
```

Comment:
Just curious, why do we have to declare this `__dummyWebpack4MainTemplate` const to get the typeof of the hooks?

Reply:
I couldn't come up with another way to get at inlined instance types.